app/services/notifications/journal_wp_notification_service.rb

```diff
@@ -45,105 +45,88 @@ def call(journal, send_notifications)
+                     # For now, we prepare the in_app reason since that one is displayed.
+                     channel_reasons['in_app'].first || channel_reasons['mail'].first,
+                     author)
       end
     end
 
     private
 
-    def create_event(journal, recipient, reason, user)
+    def create_event(journal, recipient_id, channels, reason, user)
       notification_attributes = {
-        recipient: recipient,
+        recipient_id: recipient_id,
         reason: reason,
         project: journal.project,
         resource: journal.journable,
         journal: journal,
         actor: journal.user
-      }.merge(channel_attributes(recipient, reason))
+      }.merge(channel_attributes(channels))
 
       Notifications::CreateService
         .new(user: user)
         .call(notification_attributes)
     end
 
-    def channel_attributes(recipient, reason)
-      key =
-        case reason
-        when :subscribed
-          :all
-        else
-          reason
-        end
-
-      recipient.notification_settings.map do |setting|
-        channel =
-          case setting.channel
-          when 'mail'
-            :read_email
-          when 'in_app'
-            :read_ian
-          else
-            raise "Unknown notification channel"
-          end
+    def channel_attributes(channels)
+      {
+        read_email: channels.include?('mail') ? false : nil,
+        read_ian: channels.include?('in_app') ? false : nil
 
-        [channel, setting[key] ? false : nil]
-      end.to_h
+      }
     end
 
     def notification_receivers(work_package, journal)
-      seen = mentioned(journal).each do |user|
-        yield(user, :mentioned)
-      end
-
-      seen += involved(journal, seen).each do |user|
-        yield(user, :involved)
-      end
+      receivers = receivers_hash
 
-      seen += subscribed(journal, seen).each do |user|
-        yield(user, :subscribed)
-      end
+      add_receiver(receivers, mentioned(journal), :mentioned)
+      add_receiver(receivers, involved(journal), :involved)
+      add_receiver(receivers, watched(work_package), :watched)
+      add_receiver(receivers, subscribed(journal), :subscribed)
 
-      watched(work_package, seen).each do |user|
-        yield(user, :watched)
-      end
+      receivers
     end
 
     def mentioned(journal)
       allowed_and_unique(mentioned_ids(journal),
                          journal.data.project,
-                         [])
+                         :mentioned)
     end
 
-    def involved(journal, seen)
+    def involved(journal)
       scope = User
         .where(id: group_or_user_ids(journal.data.assigned_to))
         .or(User.where(id: group_or_user_ids(journal.data.responsible)))
 
       allowed_and_unique(scope,
                          journal.data.project,
-                         seen)
+                         :involved)
     end
 
-    def subscribed(journal, seen)
+    def subscribed(journal)
```

Comment:
`subscribed` is currently present up to the user level, where a notification a user received because of this setting is marked with this word. 

This admittedly is inconsistent with the notification setting administration of users where it is called "All events".

We should fix this inconsistency on the user level and depending on the resolution chosen for it, name the method and also the rest of the code accordingly.

I am unfortunately at a loss for suitable ideas.